lib/net/http/generic_request.rb

```diff
@@ -9,6 +9,7 @@ class Net::HTTPGenericRequest
 
   def initialize(m, reqbody, resbody, uri_or_path, initheader = nil)
     @method = m
+    @retry_networking_errors = Net::HTTP::IDEMPOTENT_METHODS_.include?(m)
```

Comment:
`Net::HTTP::IDEMPOTENT_METHODS_` is only used in `lib/net/http.rb`, maybe worth yanking it from there and moving it here.